Add completion for `set-script`

Yash-Singh1 opened the pull request:
This pull request implements tab completion for `set-script`. Most of the code from `run-script`.

wraithgar committed:
fix(audit-level): add `info` audit level

This is a valid level but wasn't configured to be allowed.

Also added this param to the usage output for `npm audit`

PR-URL: https://github.com/npm/cli/pull/2923
Credit: @wraithgar
Close: #2923
Reviewed-by: @ruyadorno

Yash-Singh1 committed:
fix(set-script): add completion

PR-URL: https://github.com/npm/cli/pull/2925
Credit: @Yash-Singh1
Close: #2925
Reviewed-by: @ruyadorno

ruyadorno commented on test/lib/set-script.js:
```diff
@@ -1,7 +1,38 @@
+    t.end()
+  })
+  t.test('has package.json, with scripts', async t => {
+    writeFileSync(`${dir}/package.json`, JSON.stringify({
+      scripts: { hello: 'echo hello', world: 'echo world' },
+    }))
```
better to just use **tap** fixtures helper here instead:

```suggestion
  const localPrefix = t.testdir({
    'package.json': JSON.stringify({
      scripts: { hello: 'echo hello', world: 'echo world' },
    })
  })
  const setScript = new SetScript({ localPrefix })
    
```

ruyadorno commented on test/lib/set-script.js:
```diff
@@ -1,7 +1,38 @@
+    t.end()
+  })
+  t.test('has package.json, no scripts', async t => {
+    writeFileSync(`${dir}/package.json`, JSON.stringify({}))
```
same as L27 below

ruyadorno commented on test/lib/set-script.js:
```diff
@@ -1,7 +1,38 @@
 
+test.test('completion', t => {
+  const SetScript = requireInject('../../lib/set-script.js')
+  const dir = t.testdir()
+  const setScript = new SetScript({ localPrefix: dir })
```
it's simpler to have a fixture per test as described in the comment in L27

```suggestion
```

wraithgar committed:
fix(install): ignore auditLevel

`npm install` should not be affected by the `auditLevel` config, as the
results of audit do not change its exit status.

PR-URL: https://github.com/npm/cli/pull/2929
Credit: @wraithgar
Close: #2929
Reviewed-by: @ljharb, @ruyadorno

ruyadorno reviewed (approved):
awesome 🙌 thanks!

wraithgar committed:
fix(progress): re-add progress bar to reify

The logger was no longer in flatOptions, we pass it in explicitly now

PR-URL: https://github.com/npm/cli/pull/2944
Credit: @wraithgar
Close: #2944
Reviewed-by: @nlf

nlf committed:
fix: config should not error when workspaces are configured

PR-URL: https://github.com/npm/cli/pull/2945
Credit: @nlf
Close: #2945
Reviewed-by: @wraithgar

wraithgar committed:
fix(flatOptions): re-add `_auth`

This was not being added to flatOptions, and things like
`npm-registry-fetch` are looking for it.

PR-URL: https://github.com/npm/cli/pull/2946
Credit: @wraithgar
Close: #2946
Reviewed-by: @nlf
